src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,199 @@
+
+  /** Names of all the classes under test */
+  private Set<String> classesUnderTest = new HashSet<>();
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class BranchCoverage {
+    /** Total number of branches. */
+    public int totalBranches;
+    /** Number of uncovered branches. */
+    public int uncoveredBranches;
+  }
+
+  private CoverageTracker() {
+    this.memoryClassLoader = new MemoryClassLoader(this);
+    this.instrumenter = new Instrumenter(runtime);
+    this.data = new RuntimeData();
+
+    try {
+      this.runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Set which classes are under test.
+   *
+   * @param classesUnderTest names of all the classes under test
+   */
+  public void setClassesUnderTest(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument
+   * @return the instrumented bytes of the class, null if instrumentation failed
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!classesUnderTest.contains(className) && !isNestedClass(className)) {
+      return null;
+    }
+
+    final byte[] instrumented;
+    final String resource = '/' + className.replace('.', '/') + ".class";
+    InputStream original = getClass().getResourceAsStream(resource);
+
+    if (original == null) {
+      throw new Error("No resource with name: " + resource + " found by CoverageTracker!");
+    }
+
+    try {
+      // Instrument the class to prepare for coverage collection later.
+      instrumented = instrumenter.instrument(original, className);
+      original.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    return instrumented;
+  }
+
+  /**
+   * Determine if the given class is a nested class of a class under test.
+   *
+   * @param className the name of the class to check
+   * @return true iff the given class is a nested class of a class under test
+   */
+  private boolean isNestedClass(String className) {
+    for (String classUnderTest : classesUnderTest) {
+      if (className.startsWith(classUnderTest + "$")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Returns the instrumented version of the class.
+   *
+   * @param className name of the class
+   * @return {@code Class} object that has been instrumented for coverage data collection. Returns
+   *     null if class with target name cannot be found.
+   */
+  public Class<?> getInstrumentedClass(String className) {
+    if (className == null) {
```

Comment:
This behavior is not documented in the method specification.
Looking at the calls, they don't seem to ever pass null.  I think it would be better to remove this so that the code does not silently continue when there is a problem.